### PR TITLE
MGMT-2296 Host and cluster monitor iterate in baches

### PR DESCRIPTION
**Pull request by filanov**
monitor using batches of hosts and clusters and not load everything into memory at once

### CODE REVIEW

File: internal/host/monitor.go [hunk 1]
```diff
@@ -32,18 +32,25 @@ func (m *Manager) HostMonitoring() {
...
+			log.WithError(err).Errorf("failed to get hosts")
 			return
 		}
-		if err := m.RefreshStatus(ctx, host, m.db); err != nil {
-			log.WithError(err).Errorf("failed to refresh host %s state", *host.ID)
+		for _, host := range hosts {
+			if !m.leaderElector.IsLeader() {
+				m.log.Debugf("Not a leader, exiting HostMonitoring")
+				return
+			}
```
Comment (avishayt) [inline internal/host/monitor.go:50]: Worth adding at the start of the method too?

Comment (filanov) [reply internal/host/monitor.go:50]: There is, it's the first line in the function.

File: internal/cluster/cluster.go [hunk 2]
```diff
@@ -278,27 +282,34 @@ func (m *Manager) ClusterMonitoring() {
...
+	if err = m.db.Model(&common.Cluster{}).Where("status NOT IN (?)", noNeedToMonitorInStates).Count(&count).Error; err != nil {
 		log.WithError(err).Errorf("failed to get clusters")
 		return
 	}
-	for _, cluster := range clusters {
-		if !m.leaderElector.IsLeader() {
-			m.log.Debugf("Not a leader, exiting ClusterMonitoring")
+	for offset = 0; offset < count; offset += limit {
+		if err = m.db.Where("status NOT IN (?)", noNeedToMonitorInStates).
+			Offset(offset).Limit(limit).Order("id").Find(&clusters).Error; err != nil {
```
[the remark this answers is not in the document]
Comment (filanov) [reply internal/cluster/cluster.go:291]: Good point.

File: internal/cluster/cluster.go [hunk 3]
```diff
@@ -278,27 +282,34 @@ func (m *Manager) ClusterMonitoring() {
...
 		}
-		if clusterAfterRefresh, err = m.RefreshStatus(ctx, cluster, m.db); err != nil {
-			log.WithError(err).Errorf("failed to refresh cluster %s state", cluster.ID)
-			continue
-		}
+		for _, cluster := range clusters {
+			if !m.leaderElector.IsLeader() {
+				m.log.Debugf("Not a leader, exiting ClusterMonitoring")
+				return
+			}
```
[the remark this answers is not in the document]
Comment (filanov) [reply internal/cluster/cluster.go:298]: There it's the first line.

**Review by avishayt — APPROVED** [review]

**Pull request merged**